pkg/stanza/operator/transformer/recombine/recombine.go

```diff
@@ -344,9 +344,9 @@ func (r *Transformer) flushSource(source string, deleteSource bool) error {
 	entries := batch.entries
 
 	if r.overwriteWithOldest {
-		base = entries[0]
-	} else {
 		base = entries[len(entries)-1]
+	} else {
+		base = entries[0]
 	}
```

Comment:
@djaglowski Sorry, I don't quite understand this change. When overwrite_with = 'oldest', doesn't it mean that the attributes of the earliest entry are used? The change will use the latest entry as the base entry when overwrite_with = 'oldest'.

Reply:
You're right, I got it backwards. I've opened #32146 to correct it.